tools/test.py

```diff
@@ -375,7 +375,7 @@ def HasRun(self, output):
 
       if output.diagnostic:
         self.severity = 'ok'
-        self.traceback = output.diagnostic
+        self.traceback = '\n'.join(output.diagnostic)
```

Comment:
```suggestion
        if isinstance(output.diagnostic, list):
            self.traceback = '\n'.join(output.diagnostic)
        else:
            self.traceback = output.diagnostic
```
Better safe than sorry.

Reply:
Done!